moto/emr/models.py

```diff
@@ -501,7 +514,50 @@ def remove_tags(self, cluster_id, tag_keys):
         cluster = self.get_cluster(cluster_id)
         cluster.remove_tags(tag_keys)
 
+    def _manage_security_groups(
+        self,
+        ec2_subnet_id,
+        emr_managed_master_security_group,
+        emr_managed_slave_security_group,
+        service_access_security_group,
+        **_
+    ):
+        default_return_value = (
+            emr_managed_master_security_group,
+            emr_managed_slave_security_group,
+            service_access_security_group,
+        )
+        if not ec2_subnet_id:
+            # TODO: Set up Security Groups in Default VPC.
+            return default_return_value
+
+        try:
+            subnet = self.ec2_backend.get_subnet(ec2_subnet_id)
+        except InvalidSubnetIdError:
+            warnings.warn(
+                "Could not find Subnet with id: {0}\n"
+                "In the near future, this will raise an error.\n"
+                "Use ec2.describe_subnets() to find a suitable id "
+                "for your test.".format(ec2_subnet_id),
+                PendingDeprecationWarning,
+            )
+            return default_return_value
+
+        manager = EmrSecurityGroupManager(self.ec2_backend, subnet.vpc_id)
+        master, slave, service = manager.manage_security_groups(
+            emr_managed_master_security_group,
+            emr_managed_slave_security_group,
+            service_access_security_group,
+        )
+        return master.id, slave.id, service.id
+
     def run_job_flow(self, **kwargs):
+        if "instance_attrs" in kwargs:
```

Comment:
Do we need the if-statement? The `Instances`-param is required for the run_job_flow-method, so it looks like this kwarg will always be there